cli/lsp/documents.rs

```diff
@@ -926,9 +947,9 @@ pub struct Documents {
   /// The npm package requirements found in npm specifiers.
   npm_reqs_by_scope:
     Arc<BTreeMap<Option<ModuleSpecifier>, BTreeSet<PackageReq>>>,
-  /// Gets if any document had a node: specifier such that a @types/node package
-  /// should be injected.
-  has_injected_types_node_package: bool,
+  /// Config scopes that contain a node: specifier such that a @types/node
+  /// package should be injected.
+  scopes_with_node_specifier: Arc<HashSet<Option<ModuleSpecifier>>>,
```

Comment:
It's a little strange to see a hashmap containing an Option. Maybe we should bury this behaviour within a struct and make the public api of the struct better explain what's going on here?

Reply:
Oh, nvm, I see what this means. We may want to give `Option<ModuleSpecifier>` a name like `ScopeKey` or something?